### PR TITLE
docs: fix PHPDoc types in Database Connection.

### PR DESCRIPTION
**Pull request by ping-yee**
**Description**
See #6310 

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/Database/ConnectionInterface.php [hunk 1]
```diff
@@ -12,6 +12,8 @@
 namespace CodeIgniter\Database;
 
 /**
+ * @property Query $lastQuery
```
Comment (kenjis) [inline system/Database/ConnectionInterface.php:15]: Why does the interface have a property?

File: system/Database/BaseConnection.php [hunk 2]
```diff
@@ -456,7 +456,7 @@ abstract protected function _close();
     /**
      * Create a persistent database connection.
      *
-     * @return mixed
+     * @return false|object|resource
```
Comment (kenjis) [inline system/Database/BaseConnection.php:459]:
```suggestion
     * @return false|object|resource
     * @phpstan-return false|TConnection
```

File: system/Database/ConnectionInterface.php [hunk 3]
```diff
@@ -98,7 +100,7 @@ public function getVersion(): string;
      * Should automatically handle different connections for read/write
      * queries if needed.
      *
-     * @param mixed ...$binds
+     * @param array|string|null ...$binds
```
Comment (kenjis) [inline system/Database/ConnectionInterface.php:103]: This seems a bug.

```suggestion
     * @param array|string|null $binds
```

File: system/Database/SQLSRV/Connection.php [hunk 4]
```diff
@@ -44,7 +44,7 @@ class Connection extends BaseConnection
      * FALSE or SQLSRV_CURSOR_FORWARD would increase performance,
      * but would disable num_rows() (and possibly insert_id())
      *
-     * @var mixed
+     * @var bool|string
```
Comment (kenjis) [inline system/Database/SQLSRV/Connection.php:47]: Can be `true`?
```suggestion
     * @var false|string
```

File: system/Database/ConnectionInterface.php (hunk 1 above)
Comment (ping-yee) [reply system/Database/ConnectionInterface.php]: I have been removed.

File: system/Database/BaseConnection.php [hunk 5]
```diff
@@ -558,7 +558,7 @@ abstract protected function execute(string $sql);
      * Should automatically handle different connections for read/write
      * queries if needed.
      *
-     * @param mixed ...$binds
+     * @param array|string|null ...$binds
```
[the remark this answers is not in the document]
Comment (ping-yee) [reply system/Database/BaseConnection.php]: I didn't modify the property name before.
I have been removed.

File: system/Database/ConnectionInterface.php (hunk 3 above)
Comment (ping-yee) [reply system/Database/ConnectionInterface.php]: Is there anything need to be fixed besides removing `...`?

File: system/Database/SQLSRV/Connection.php (hunk 4 above)
Comment (ping-yee) [reply system/Database/SQLSRV/Connection.php]: I have been modified.

File: system/Database/ConnectionInterface.php (hunk 3 above)
Comment (kenjis) [reply system/Database/ConnectionInterface.php]: No, for now.

The difference of the parameter count may be a bug.

**Review by kenjis — APPROVED** [review]
Thank you!

**Pull request merged**